File: lib/Model/DeviceModel.dart
import 'package:fble/Common/APIConsts.dart';

class DeviceModel {
  int status = 0;
  String name = '';
  String address = '';
  bool isScreenSharing = false;
  bool isRegistered = false;
  bool isConnecting = false;

  DeviceModel();

  factory DeviceModel.fromJSON(Map<String, dynamic> res) {
    DeviceModel model = new DeviceModel();

    model.name = res[APIConsts.name];
    model.address = res[APIConsts.address];

    return model;
  }

  List<DeviceModel> getList(List<dynamic> res) {
    List<DeviceModel> allData = [];
    res.forEach((element) {
      DeviceModel model = DeviceModel.fromJSON(element);
      allData.add(model);
    });

    return allData;
  }
}

File: android/app/src/main/java/com/sts/fble/Model/DeviceModel.java
package com.sts.fble.Model;

import com.sts.fble.Common.APIConsts;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.Serializable;

public class DeviceModel implements Serializable {
    public String name = "";
    public String address = "";

    public DeviceModel(String name, String address) {
        this.name = name;
        this.address = address;
    }

    public DeviceModel(JSONObject json) {
        try {
            name = json.getString(APIConsts.name);
            address = json.getString(APIConsts.address);
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    public JSONObject toJSON(){
        JSONObject object = new JSONObject();
        try {
            object.put(APIConsts.name, name);
            object.put(APIConsts.address, address);
        } catch (JSONException e) {
            e.printStackTrace();
        }

        return  object;
    }
}
